modules/cats-effect/src/main/scala/pureconfig/module/catseffect/package.scala

```diff
@@ -169,42 +137,39 @@ package object catseffect {
     *
     * @param conf The configuration to save
     * @param outputPath Where to write the configuration
-    * @param blocker the blocking context which will be used to load the configuration.
     * @param overrideOutputPath Override the path if it already exists
     * @param options the config rendering options
     * @return The return action will save out the supplied configuration upon invocation
     */
   def blockingSaveConfigAsPropertyFileF[F[_], A](
       conf: A,
       outputPath: Path,
-      blocker: Blocker,
       overrideOutputPath: Boolean = false,
       options: ConfigRenderOptions = ConfigRenderOptions.defaults()
-  )(implicit F: Sync[F], csf: ContextShift[F], writer: ConfigWriter[A]): F[Unit] = {
-    val fileAlreadyExists =
+  )(implicit F: Sync[F], writer: ConfigWriter[A]): F[Unit] = {
+    val fileAlreadyExists: F[Unit] =
       F.raiseError(
         new IllegalArgumentException(s"Cannot save configuration in file '$outputPath' because it already exists")
       )
 
-    val fileIsDirectory =
+    val fileIsDirectory: F[Unit] =
       F.raiseError(
         new IllegalArgumentException(s"Cannot save configuration in file '$outputPath' because it is a directory")
       )
 
     val check =
-      F.defer {
-        if (!overrideOutputPath && Files.isRegularFile(outputPath)) fileAlreadyExists
-        else if (Files.isDirectory(outputPath)) fileIsDirectory
-        else F.unit
-      }
+      F.ifElseM(
```

Comment:
I would recommend `F.blocking(...).ifM(ifTrue = ..., ifFalse = ...)` instead, more explicit

Reply:
Thanks! Didn't think about syntax version of `ifM`. Fits really nice here.